scripts/quests/otherAreas/Knocking_on_Forbidden_Doors.lua

```diff
@@ -0,0 +1,296 @@
+
+            onEventFinish =
+            {
+                [536] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+
+        [xi.zone.PHOMIUNA_AQUEDUCTS] =
+        {
+            ['Wooden_Ladder'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        quest:getVar(player, 'Prog') == 1 and
+                        npc:getID() == phomiunaID.npc.LADDER_KNOCKING
+                    then -- Find a better way to identify exactly which ladder it is in case IDs shift
+                        return quest:progressEvent(38, 27) -- Essentially chains the ladder cs with the actual quest cs
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [38] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+            },
+        },
+
+        [xi.zone.MISAREAUX_COAST] =
+        {
+            ['qm_mire_incense'] =
+            {
+                onTrigger = function(player, npc)
+                    --- ??? by end of river
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        not player:hasKeyItem(xi.ki.MIRE_INCENSE)
+                    then
+                        return quest:keyItem(xi.ki.MIRE_INCENSE)
+                    end
+                end,
+            },
+
+            ['qm_alsha'] =
+            {
+                onTrigger = function(player, npc)
+                    local progressVar = quest:getVar(player, 'Prog')
+                    -- First time clicking on ???
+                    if
+                        player:hasKeyItem(xi.ki.MIRE_INCENSE) and
+                        progressVar == 2
+                    then
+                        return quest:progressEvent(556, { [1] = xi.ki.MIRE_INCENSE })
+
+                    -- Second time clicking on ???
+                    elseif progressVar == 3 then
+                        return quest:progressEvent(557, { [1] = xi.ki.MIRE_INCENSE })
+
+                    -- Clicking on the ??? after killing NM
+                    elseif progressVar == 4 then
+                        return quest:event(558)
+                    end
+                end,
+            },
+
+            ['Alsha'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        quest:setVar(player, 'Prog', 4)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [556] = function (player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+
+                [557] = function (player, csid, option, npc)
+                    if quest:getVar(player, 'Prog') == 3 and
```

Comment:
Looks like I missed one formatting spot in the last review, multiline conditional

Reply:
Just pushed a fix on this one. 

Many thanks for y'alls patience on this one.  I've definitely learned a ton here - very much appreciated :)